From 42518209c039862f32068aeee497621aaac776ed Mon Sep 17 00:00:00 2001
From: Xu Chenjiao <xuchenjiao@wxiat.com>
Date: Tue, 31 May 2022 14:11:10 +0800
Subject: [PATCH] sw64: fix compile errors for NOT chip3
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 42518209c039862f32068aeee497621aaac776ed
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56QAM

--------------------------------

IO register DEVINT_WKEN, DEVINTWK_INTEN, PME_ENABLE_INTD_CORE0 and
AER_ENABLE_INTD_CORE0 are only defined in chip3. This will lead to
compile errors if other chip is selected. To avoid these errors, we
define default set_devint_wken() and set_pcieport_service_irq() as
weak symbols.

Signed-off-by: Xu Chenjiao <xuchenjiao@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/chip/chip3/chip.c | 16 ++++++++++++++++
 arch/sw_64/kernel/pci.c      | 15 ++++-----------
 2 files changed, 20 insertions(+), 11 deletions(-)

diff --git a/arch/sw_64/chip/chip3/chip.c b/arch/sw_64/chip/chip3/chip.c
index acd25fe99669..b73afc9c62ac 100644
--- a/arch/sw_64/chip/chip3/chip.c
+++ b/arch/sw_64/chip/chip3/chip.c
@@ -90,6 +90,22 @@ void setup_chip_clocksource(void)
 #endif
 }
 
+void set_devint_wken(int node)
+{
+	unsigned long val;
+
+	/* enable INTD wakeup */
+	val = 0x80;
+	sw64_io_write(node, DEVINT_WKEN, val);
+	sw64_io_write(node, DEVINTWK_INTEN, val);
+}
+
+void set_pcieport_service_irq(int node, int index)
+{
+	write_piu_ior0(node, index, PMEINTCONFIG, PME_ENABLE_INTD_CORE0);
+	write_piu_ior0(node, index, AERERRINTCONFIG, AER_ENABLE_INTD_CORE0);
+}
+
 static int chip3_get_cpu_nums(void)
 {
 	unsigned long trkmode;
diff --git a/arch/sw_64/kernel/pci.c b/arch/sw_64/kernel/pci.c
index 401ee0b781be..a004ea9fde7f 100644
--- a/arch/sw_64/kernel/pci.c
+++ b/arch/sw_64/kernel/pci.c
@@ -600,15 +600,7 @@ sw64_init_host(unsigned long node, unsigned long index)
 	}
 }
 
-static void set_devint_wken(int node)
-{
-	unsigned long val;
-
-	/* enable INTD wakeup */
-	val = 0x80;
-	sw64_io_write(node, DEVINT_WKEN, val);
-	sw64_io_write(node, DEVINTWK_INTEN, val);
-}
+void __weak set_devint_wken(int node) {}
 
 void __init sw64_init_arch(void)
 {
@@ -651,6 +643,8 @@ void __init sw64_init_arch(void)
 	}
 }
 
+void __weak set_pcieport_service_irq(int node, int index) {}
+
 static void __init sw64_init_intx(struct pci_controller *hose)
 {
 	unsigned long int_conf, node, val_node;
@@ -679,8 +673,7 @@ static void __init sw64_init_intx(struct pci_controller *hose)
 	if (sw64_chip_init->pci_init.set_intx)
 		sw64_chip_init->pci_init.set_intx(node, index, int_conf);
 
-	write_piu_ior0(node, index, PMEINTCONFIG, PME_ENABLE_INTD_CORE0);
-	write_piu_ior0(node, index, AERERRINTCONFIG, AER_ENABLE_INTD_CORE0);
+	set_pcieport_service_irq(node, index);
 }
 
 void __init sw64_init_irq(void)
-- 
2.33.0

